src/celeritas/em/distribution/WentzelDistribution.hh

```diff
@@ -0,0 +1,309 @@
+            const real_type x = sqrt(2 * inc_mom_sq() * (1 - cos_t)) * ccoef
+                                * 2;
+            return flat_form_factor(x)
+                   * flat_form_factor(
+                       x * 0.6
+                       * fastpow(value_as<Mass>(target_.nuclear_mass()),
+                                 static_cast<real_type>(1) / 3));
+        }
+        break;
+        case NuclearFormFactorType::Exponential:
+            return 1 / ipow<2>(1 + formf * (1 - cos_t));
+        case NuclearFormFactorType::Gaussian:
+            return exp(-2 * formf * (1 - cos_t));
+        default:
+            return 1;
+    }
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Flat form factor
+ * TODO: Reference?
+ */
+CELER_FUNCTION real_type WentzelDistribution::flat_form_factor(real_type x) const
+{
+    return 3 * (sin(x) - x * cos(x)) / ipow<3>(x);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Atomic number of the target isotope
+ */
+CELER_FUNCTION int WentzelDistribution::target_Z() const
+{
+    return target_.atomic_number().get();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Helper function to calculate the incident momentum squared.
+ */
+CELER_FUNCTION real_type WentzelDistribution::inc_mom_sq() const
+{
+    return inc_energy_ * (inc_energy_ + 2 * inc_mass_);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculates the Moilere screening coefficient.
+ */
+CELER_FUNCTION real_type WentzelDistribution::compute_screening_coefficient() const
+{
+    // TODO: Reference for just proton correction?
+    real_type correction = 1;
+    const real_type sq_cbrt_z = fastpow(static_cast<real_type>(target_Z()),
+                                        static_cast<real_type>(2) / 3);
+    if (target_Z() > 1)
+    {
+        const real_type tau = inc_energy_ / inc_mass_;
+        // TODO: Reference for this factor?
+        const real_type factor = sqrt(tau / (tau + sq_cbrt_z));
+        const real_type inv_beta_sq = 1 + ipow<2>(inc_mass_) / inc_mom_sq();
+
+        correction = min(
+            target_Z() * 1.13,
+            1.13
+                + 3.76 * ipow<2>(target_Z() * constants::alpha_fine_structure)
+                      * inv_beta_sq * factor);
+    }
+
+    return correction * value_as<MomentumSq>(data_.screen_r_sq_elec)
+           * sq_cbrt_z / inc_mom_sq();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculates the (cosine of the) maximum polar angle that incident particle
+ * can scatter off of the target's electrons.
+ */
+CELER_FUNCTION real_type WentzelDistribution::compute_max_electron_cos_t() const
+{
+    // TODO: Need to validate against Geant4 results
+    const real_type max_energy = is_electron_ ? inc_energy_ / 2 : inc_energy_;
```

Comment:
```suggestion
    real_type max_energy = is_electron_ ? real_type(0.5) * inc_energy_  :  inc_energy_;
```

Reply:
No strict rule here, but better for the performance point of view (i.e., * is cheaper than /).